homie-controller/src/types.rs

```diff
@@ -167,6 +187,64 @@ impl Property {
     pub fn has_required_attributes(&self) -> bool {
         self.name.is_some() && self.datatype.is_some()
     }
+
+    pub fn value<T: Value>(&self) -> Result<T, ValueError> {
```

Comment:
I'm a little bit uncomfortable with having a function that has the same name as a property. Looks like the compiler is able to work out what you mean though, so maybe it's fine?

Reply:
What would you suggest calling it instead, is there some typical naming convention?

Reply:
Honestly not sure. I will ask at work. Might be that your approach is how everyone does it.

Reply:
Ah. Apparently it is common, but normally the field is private.

Reply:
Okay, I'll keep it as is.